### ISSUE TITLE
Describe ECH chronologically rather than all client behavior then all server behavior

### ISSUE THREAD

**Issue by cbartle891** [tlswg/draft-ietf-tls-esni#412]
Imo, it's very confusing to describe all the client behavior and then all the server behavior as opposed to describing the steps in the order in which they happen. I'm constantly flipping back and forth from one section to the other to understand the series of events. 

@cjpatton [suggested](https://github.com/tlswg/draft-ietf-tls-esni/pull/407#discussion_r604270551) that the current method of organization might be easier for implementers, but I'm not so sure. It seems to me that the first step in implementing something is understanding it, and it's more difficult to understand ECH in its current format than if it were explained in the order in which things happen.

@dmcardle do you think that would be more confusing or less so?

**Comment by dmcardle** [comment]
I can second that experience — I've found myself flipping back-and-forth between the behavior sections.  On the other hand, I've also found it useful to step through just the server behavior section and compare it with my server-only patch.

I'm slightly concerned that weaving the sections together would add a lot of verbal overhead when switching between the client and server. But I wouldn't object if you or someone else wanted to try to refactor it :)
